### PR TITLE
[db] Add organisationId to d_b_oidc_client_config + indices

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Part of https://github.com/gitpod-io/gitpod/issues/15708

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/gitpod-db/src/typeorm/migration/1673944879912-AddOrganisationIDToOIDCClientConfig.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                `ALTER TABLE ${table} ADD COLUMN ${column} varchar(255), ALGORITHM=INPLACE, LOCK=NONE`,
+            );
+        }
+
+        if (!(await indexExists(queryRunner, table, orgIndex))) {
+            await queryRunner.query(`CREATE INDEX \`${orgIndex}\` ON \`${table}\` (${column})`);
+        }
+
+        if (!(await indexExists(queryRunner, table, idOrgCompositeIndex))) {
+            await queryRunner.query(`CREATE INDEX \`${idOrgCompositeIndex}\` ON \`${table}\` (id, ${column})`);
```
Comment (AlexTugarev) [inline components/gitpod-db/src/typeorm/migration/1673944879912-AddOrganisationIDToOIDCClientConfig.ts:28]: 💡 Idea, I assume `idx_id_organisationId` is optimizing for queries where we'd use `id` and `orgId` to fetch a single entry.

`id` should be already unique within the system. When trying to read the client config from DB we should already have the `id` at hand, which can be used for direct access. It's not obvious to see the benefit of `idx_id_organisationId`.

Comment (easyCZ) [reply components/gitpod-db/src/typeorm/migration/1673944879912-AddOrganisationIDToOIDCClientConfig.ts:28]: The reason is the same as in https://github.com/gitpod-io/gitpod/pull/15777#discussion_r1072246207, the lookup should be by the tuple (id, org_id).

Comment (easyCZ) [reply components/gitpod-db/src/typeorm/migration/1673944879912-AddOrganisationIDToOIDCClientConfig.ts]: @AlexTugarev Do you want me to drop the extra index?

**Review by AlexTugarev — APPROVED** [review]

**Pull request merged**